### ISSUE TITLE
HacMini Update Corrupts OpenCore

### ISSUE THREAD

**Issue by spillinova** [osy86/HaC-Mini#940]
**Describe the bug**
Attempted to update HaCMini version from within MacOS 10.14.6 from HaCMini 2.15 to HaCMini 3.1.1.  After successful install OpenCore boot menu disappeared upon reboot.

**Expected behavior**
Expected to run the update and then reboot back into MacOS.

**Environment:**
 - Running OSX version: MacOS 10.14
 - Target drive OSX version: MacOS 10.14
 - HaC Mini installer version: 3.1.1

**Additional context**

- Update to 3.1.1
	- Installed update to main NUC8i7HVB machine running MacOS 10.14.6
		- Original install was dual boot to BootCamp on partitioned SSD
	- After reboot OpenCore boot manager was gone.
		- Windows and clear NVRAM remained - only Windows Boot Manager remained?
		- No way to boot to MacOS
		- Windows not booting when I select it
		- After selecting clear NVMRAM was able to successfully load Windows
- Referred to Issue #47
	- Disabled "Secure Boot"
- Referred to Issue #134
	- Cleared Bios Password by removing the jumper 
	- Recovered the bios using the newest version bio - HNKBLi70.0067
- Got desperate and decided to wipe the drive (didn't care about having BootCamp anyway...plus I wanted to recover some space on my SSD and upgrade to Catalina)
	- Used “Clean All” from DISKPART on Windows Install USB
	- Created 2 USB HacMini Installers on alternate machine running MacOS 10.14.6
		- USB 2.0 ScanDisk Cruzer with Catalina and HaCMini 3.1.1 patch (no Secure Boot option)
			- Single front USB 2.0 port
			- Doesn't finish MacOS installation (Circles back as a new install after reboot)
				- The installer is not creating an EFI folder in the SSD EFI partition
		- USB 3.0 ScanDisk Cruzer with Catalina and HaCMini 3.1.1 patch (no Secure Boot option)
			- Back USB Ports
			- With USB 3.0 HacMini Installer  Bios shows OpenCore USB but won't boot to it (refer to image 1)
				- Was able to boot to USB 3.0 MacMini installer via bootx64.efi from EFI Shell
					- Still doesn't finish MacOS installation (Circles back as a new install after reboot)
						- The installer is not creating an EFI folder in the SSD EFI partition (refer to image 2)
	- Created Windows installer on USB 3.0 ScanDisk Cruzer
		- Bios Shows Part 0 OpenCore and Windows OS Boot Loader but won't boot to OpenCore
		- Windows USB Installer is bootable from all USB ports

![IMG_0868](https://user-images.githubusercontent.com/5526129/136296362-fd884c9c-a2e0-419d-b130-9ecaeeb968b9.jpg)

![IMG_9289](https://user-images.githubusercontent.com/5526129/136296527-4c8e0e54-dc8c-465c-85dc-78384c6a6051.jpg)

**Logs**
No install log - no files exist after initial install attempts

**Comment by densimpson** [comment]
I have the same issue as well
a clean install with latest OSX 10.15 and HaCMini 3.1.1

just like you 
Still doesn't finish MacOS installation (Circles back as a new install after reboot)
The installer is not creating an EFI folder in the SSD EFI partition 
and 
The Opencore bootmenu just show the USB Installer and Reset NVRAM option in the first boot after install.

SSD: Samsung 960 Pro 512GB
Bios: HN0067
Hacmini 3.11 with default settings except:

enable Patch OSX installer
enable Verbose boot
enable Debug mode
enable Show boot menu

**Comment by spillinova** [comment]
Here are the steps I just took:
1) Ran "Clean All" from DISKPART 3 times in a row to make sure the SSD has no Windows EFI references (see image 1)
2) Restored the Bios from bio file -  HNKBLi70.0067
3) Cleared Bios Passwords by removing the jumper
4) Booted to Bios and set 3 needed settings for HacMini
5) Rebooted to OpenCore Boot Menu and Cleared NVMRAM
6) Rebooted to OpenCore Boot Menu and Ran Catalina Install from USB 2.0 with HaCMini patch and "boot menu" option
7) Ran MacOS Disk Manager and reformatted SSD to APFS with GUID
8) Began Install of Catalina on newly formatted SSD
9) During installation the computer reboots 1 time on its own and appears to continue installation, but then circles back to the new install screen (see image 3)
10) Clicked on install again (to the same drive) and the install just repeats step 9 above
11) Booted into EFI shell and checked FS0 (SSD) - **No files or directories** (see image 2)

![IMG_9313](https://user-images.githubusercontent.com/5526129/136620179-7f4dd8d1-6db5-41f7-a935-564b83335c3c.JPG)

![IMG_9312](https://user-images.githubusercontent.com/5526129/136620131-afa711cf-48f0-499f-87df-c14d9b48acdc.JPG)

![IMG_9315](https://user-images.githubusercontent.com/5526129/136620462-0df73888-ab24-443d-ba6d-98884f07005f.JPG)

**Comment by HorstBrack75** [comment]
In step 9, if you reboot from your patched installer USB, don’t you see several entries from the installation in progress?

**Comment by spillinova** [comment]
I get the Catalina installer and Clear NVRAM.  If I choose nothing it will go back into a new install, and if I choose the Catalina installer it will go back into a new install.

**Comment by HorstBrack75** [comment]
Your installer is 10.14.5+?

What SSD(s) are you using?

**Comment by spillinova** [comment]
Installer is 10.15 and it was created on a machine running 10.14.6
The SSD is SABRENT (SB-ROCKET-1TB)

**Comment by spillinova** [comment]
Yes. Bios is the default bios settings, except for the 3 changes that the HaCMini instructions require.  It's not installing to the USB because the install begins writing files to the SSDs main partition. (for instance the log file I attached to my first post was found on the SSD).  When I start up with F10 I get the boot menu and the only option is OpenCore.  When I click on it the screen flashed for a second and then goes back to the boot menu.  I can just keep clicking on it endlessly.

**Comment by HorstBrack75** [comment]
I think your ESP partition is either corrupted or is not correct. The guide specifically asks you to format your drive and install Windows AFTER OS X.

I suggest, you fill in the issue template with all the information and hopefully  somebody else will chime in.

**Comment by spillinova** [comment]
@HorstBrack75 I appreciate you trying to help,  but you're not actually paying attention to what I wrote in my first post. I had a dual boot setup on my nook. I did it by installing Mojave and HacMini 2.15. Then I installed BootCamp Windows. I've been running that set up for almost a year. Then I ran the installer for HaCMini 3.1.1, but missed the special instructions, so when I rebooted OpenCore was gone and only Windows remained. So I wiped the disk and now the only thing I can install on this NUC is Windows. If I install Windows, the EFI partition is successfully created, but its a Windows EFI. If I wipe the drive and run the Catalina installer with the HaCMini patch and boot menu arg, the install will never complete and there will be no files or folders in the EFI partition.
